File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.py
import sys
input = sys.stdin.readline

S = input().rstrip()
S_set = set()
i = 0
j = 1

while i < len(S):
    while j < len(S)+1:
        S_set.add(S[i:j])
        j += 1
    i += 1
    j = i+1

print(len(S_set))

File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim()
// let input = fs.readFileSync("./EX.txt").toString().trim()

const S = input.trim()
let S_set = new Set()

let i = 0
let j = 1

while (i < S.length) {
    while(j < S.length+1) {
        S_set.add(S.slice(i, j))
        j ++
    }
    i++
    j = i+1
}

console.log(S_set.size)